com.unity.render-pipelines.universal/Runtime/RenderingUtils.cs

```diff
@@ -154,27 +158,34 @@ internal class StereoConstants
         /// <param name="cmd">CommandBuffer to submit data to GPU.</param>
         /// <param name="viewMatrix">View matrix to be set. Array size is 2.</param>
         /// <param name="projectionMatrix">Projection matrix to be set.Array size is 2.</param>
+        /// <param name="cameraProjectionMatrix">Camera projection matrix to be set.Array size is 2. Does not include platform specific transformations such as depth-reverse, depth range in post-projective space and y-flip. </param>
         /// <param name="setInverseMatrices">Set this to true if you also need to set inverse camera matrices.</param>
         /// <returns>Void</c></returns>
-        internal static void SetStereoViewAndProjectionMatrices(CommandBuffer cmd, Matrix4x4[] viewMatrix, Matrix4x4[] projMatrix, bool setInverseMatrices)
+        internal static void SetStereoViewAndProjectionMatrices(CommandBuffer cmd, Matrix4x4[] viewMatrix, Matrix4x4[] projMatrix, Matrix4x4[] cameraProjMatrix, bool setInverseMatrices)
```

Comment:
Should we split cameraProjMatrix and per-pass view, proj matrix as they have different update frequencies?

Reply:
cameraProjMatrix is maybe misnamed. It is actually the projMatrix, but before it is modified by `GL.GetGPUProjectionMatrix()`.
Should I rename:
  projMatrix => gpuProjMatrix,
  cameraProjMatrix => projMatrix
?

Reply:
Sorry, missed the comment. So from shaders there are two distinction for projection matrices, per-pass (UNITY_MATRIX_P) and per-camera (CameraProjectionMatrix). These might be the same for a given pass, but not always, for example, in shadow caster pass you have UNITY_MATRIX_P to be lighting projection matrix while CameraProjectionMatrix is always current rendering camera projection matrix regardless of the pass.

For API point of view, it seems confusing we are passing both a projection and a camera projection, thus why my question. Why not having SetViewAndProjectionMatrices / SetCameraMatrices separate functions? (I think that's what we do for non stereo path?)

Reply:
yep, renaming it would be more clear I think.

Reply:
Hi I started making the change but realised it does not solve ambiguity (make it worse on the call site).
I think we should keep this naming convention as it, because it matches the naming convention in the shader code (proj matrix means it it has been modified by GL.GetGPUProjectionMatrix(), while cameraProj implies it is a "true" non-generic projection matrix).

Reply:
ok. agreed.